omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/VisitConfigurationController2_0.java

```diff
@@ -0,0 +1,180 @@
+	}
+
+	private Boolean getEnableVisitsValue(AdministrationService administrationService) {
+		String enableVisits = administrationService
+				.getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_ENABLE_VISITS, "true");
+		return Boolean.parseBoolean(enableVisits);
+	}
+
+	private Boolean getAutoCloseVisitsTaskStartedValue(SchedulerService schedulerService) {
+		TaskDefinition autoCloseVisitsTaskStarted = schedulerService
+				.getTaskByName(OpenmrsConstants.AUTO_CLOSE_VISITS_TASK_NAME);
+
+		if (autoCloseVisitsTaskStarted != null) {
+			return autoCloseVisitsTaskStarted.getStarted();
+		}
+
+		return false;
+	}
+
+	private void updateGetAutoCloseVisitsTaskStartedValue(SchedulerService schedulerService,
+			Boolean autoCloseVisitsTaskStarted) throws SchedulerException {
+		TaskDefinition closeVisitsTask = schedulerService.getTaskByName(OpenmrsConstants.AUTO_CLOSE_VISITS_TASK_NAME);
+		if (closeVisitsTask != null) {
+			if (autoCloseVisitsTaskStarted && !closeVisitsTask.getStarted()) {
+				schedulerService.scheduleTask(closeVisitsTask);
+			} else if (!autoCloseVisitsTaskStarted && closeVisitsTask.getStarted()) {
+				schedulerService.shutdownTask(closeVisitsTask);
+			}
+		}
+	}
+
+	private List<VisitType> getVisitTypesToAutoCloseValue(AdministrationService administrationService,
+			VisitService visitService) {
+		String gpValue = administrationService.getGlobalProperty(OpenmrsConstants.GP_VISIT_TYPES_TO_AUTO_CLOSE);
+		if (StringUtils.isNotBlank(gpValue)) {
+			List<VisitType> visitTypes = new ArrayList<>();
+			String[] visitTypeNames = StringUtils.split(gpValue.trim(), ",");
+			for (int i = 0; i < visitTypeNames.length; i++) {
+				String currName = visitTypeNames[i];
+				visitTypeNames[i] = currName.trim().toLowerCase();
+			}
+
+			List<VisitType> allVisitTypes = visitService.getAllVisitTypes();
+			for (VisitType visitType : allVisitTypes) {
+				if (ArrayUtils.contains(visitTypeNames, visitType.getName().toLowerCase())) {
+					visitTypes.add(visitType);
+				}
+			}
+			return visitTypes;
+		}
+		return Collections.emptyList();
+	}
+
+	private void updateVisitTypesToAutoCloseValue(AdministrationService administrationService, VisitService visitService,
+			List<VisitType> visitTypesToAutoClose) {
+		// visitTypesToAutoClose contains only uuids, map to full visit types
+		List<String> visitTypesToAutoCloseUuids = visitTypesToAutoClose.stream().map(BaseOpenmrsObject::getUuid)
+				.collect(Collectors.toList());
+		List<VisitType> visitTypesToAutoCloseFull = getVisitTypesByUuids(visitTypesToAutoCloseUuids, visitService);
+
+		String visitTypeNames = visitTypesToAutoCloseFull.stream().map(BaseOpenmrsMetadata::getName)
+				.collect(Collectors.joining(","));
+
+		GlobalProperty visitTypesToAutoCloseGp = new GlobalProperty(OpenmrsConstants.GP_VISIT_TYPES_TO_AUTO_CLOSE,
+				visitTypeNames);
+		administrationService.saveGlobalProperty(visitTypesToAutoCloseGp);
```

Comment:
Why not `setGlobalProperty(String propertyName, String propertyValue)`?

Reply:
https://github.com/openmrs/openmrs-module-webservices.rest/pull/489/commits/58171b8ee25befca65f4449686ad108c761ff5d4